Content.Server/_Moffstation/Players/WatchListTracker.cs

```diff
@@ -0,0 +1,59 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Content.Server.Administration.Notes;
+using Content.Server.Database;
+using Robust.Server.Player;
+using Robust.Shared.Enums;
+using Robust.Shared.Player;
+
+namespace Content.Server._Moffstation.Players;
+
+/// <summary>
+/// This is a class that allows synchronous code to identify whether a player is WatchListed
+/// </summary>
+public sealed class WatchListTracker : EntitySystem
+{
+    [Dependency] private readonly IAdminNotesManager _notes = default!;
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+
+    public override void Initialize()
+    {
+        _playerManager.PlayerStatusChanged += OnPlayerStatusChanged;
+    }
+
+    private readonly Dictionary<ICommonSession, bool> _watchLists = new();
```

Comment:
Instead of a dictionary, can this just be a `Set` where presence in the set means the session is watchlisted, and absence means it's not?